### PR TITLE
Spencer's "final" editorial pass

### PR DESCRIPTION
**Pull request by SpencerDawkins**
This was a pretty careful editorial pass. I also deleted the Terms Used section - I think we're good.

### CODE REVIEW

**Comment by SpencerDawkins** [comment]
I should point out that we still have seven other PRs that haven't been merged yet. I'm participating in several of those discussions, but didn't include them in my "final" review yet.

File: draft-quic-atsss-reqs.mkd [hunk 1]
```diff
@@ -245,28 +246,27 @@ Also, the document uses the following additional terms:
 
 # High level Overview {#overview}
 
-The 5G Core (5GC) supports a service that provides exchange of data between a User Equipment (UE) and a data network (referred to as Packet Data Network (PDN)) identified by a Data Network Name (DNN). This connectivity service, called PDU Connectivity Service, is realized via 'PDU Sessions' that are established upon request from the UE when the UE first connects to that network. The type of PDU Session can be IPv4, IPv6, IPv4v6, Ethernet or Unstructured.
+The 5G Core (5G Core) supports a service that provides exchange of data between a User Equipment (UE) and a data network (referred to as Packet Data Network (PDN)) identified by a Data Network Name (DNN). This connectivity service, called the Protocol Data Unit Connectivity Service, is realized via 'PDU Sessions' that are established upon request from User Equipment (UE) when the UE first connects to that network. The type of PDU Session can be IPv4, IPv6, IPv4v6, Ethernet or Unstructured.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
The 5G Core supports a service that provides exchange of data between a User Equipment (UE) and a data network (referred to as Packet Data Network (PDN)) identified by a Data Network Name (DNN). This connectivity service, called the Protocol Data Unit (PDU) Connectivity Service, is realized via 'PDU Sessions' that are established upon request from User Equipment (UE) when the UE first connects to that network. The type of PDU Session can be IPv4, IPv6, IPv4v6, Ethernet or Unstructured.
```

File: draft-quic-atsss-reqs.mkd [hunk 2]
```diff
@@ -245,28 +246,27 @@ Also, the document uses the following additional terms:
...
+It is out of the scope of this document to provide a comprehensive overview of 5G System (5GS) architecture. In particular, this document does not describe how PDU Sessions are established, and thus how IP addresses/prefixes are assigned to requesting UEs. 
 
-An UE can be provided a multi-access PDU Connectivity Service. That is, an UE can exchange data with a PDN by using a "3GPP access network" and a "non-3GPP access network". This is realized by use of the ATSSS service that is provided in the 5G core network usually collocated with the User Plane Function (UPF) that manages the UE's PDU session.
+An UE can be provided a multi-access PDU Connectivity Service. That is, an UE can exchange data with a PDN by using a "3GPP access network" and a "non-3GPP access network" (often a WLAN that is not managed by the 5G network provider). This is realized by use of the ATSSS service that is provided in the 5G core network usually collocated with the User Plane Function (UPF) that manages the UE's PDU session.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
An UE can be provided a multi-access PDU Connectivity Service. That is, an UE can exchange data with a PDN by using a "3GPP access network" and a "non-3GPP access network" (often a WLAN). This is realized by use of the ATSSS service that is provided in the 5G core network usually collocated with the User Plane Function (UPF) that manages the UE's PDU session.
```

File: draft-quic-atsss-reqs.mkd [hunk 3]
```diff
@@ -428,94 +428,94 @@ Legend:
...
 
-- to avoid frequent connection migration if both access networks could be used to forward data (each with a distinct source IP address).
+- avoiding frequent connection migration if both access networks could be used to forward packets (each with a distinct source IP address).
 
 ## QUIC as an ATSSS Data Plane Protocol {#sec-quic-data}
 
 This section elaborates how non-TCP traffic (UDP or a subset like QUIC) can to be encapsulated into a tunnel between the UE and the ATSSS UPF to enable so called ATSSS higher-layer steering for all traffic, not only TCP or QUIC. This section discusses to what extent QUIC can be used as a tunneling protocol for the ATSSS service and whether gaps are found.
 
-When tunneling non-TCP (e.g., UDP, plain IP) over QUIC the Unreliable Datagram Extension
+When tunneling non-TCP (e.g., UDP, IP, etc.) over QUIC the Unreliable Datagram Extension
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
When tunneling non-TCP (e.g., UDP, IP) over QUIC the Unreliable Datagram Extension
```

"Etc." is not an example ;-)

File: draft-quic-atsss-reqs.mkd [hunk 4]
```diff
@@ -428,94 +428,94 @@ Legend:
...
+induces a delay when switching as packets are buffered. Further, congestion control state is
+reset on a new path and needs to ramp up slowly. When frequent handovers happen, 
 splitting traffic over multiple paths simulaneously can be beneficial for a smooth user experience.
 Further, of course, the ability to use multiple paths simultaneously also increases the 
 maximum capacity available, which can also be benefical in some cases.
 
+<<<<<<< Updated upstream
 This option is not a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
+=======
+This option is not considered a valid approach for the ATSSSv2 discussed within the 3GPP.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
This option is not considered a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
```

File: draft-quic-atsss-reqs.mkd [hunk 5]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
 since data from the same flow are sent over different connections,
 again using unreliable datagram to minimise head-of-line blocking.
 Because both (tunnel) QUIC connections are completely independent of each other and the paths on the
-different access networks may have different latency, this approach would
-results in reordering of packets that belong to
+different access networks may have different latency, this approach would likely
+result in reordering of packets that belong to
 a split flow. If reordering should be avoided, this would require additional signaling
-between the UE and the ATSSS UPF, e.g., adding sequence numbers, and a reordering buffer and logic.
+between the UE and the ATSSS UPF, e.g., adding sequence numbers, and adding a reordering buffer and logic to both the UE and ATSSS UDF.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
between the UE and the ATSSS UPF, e.g., adding sequence numbers, and adding a reordering buffer and logic to both the UE and ATSSS UPF.
```

File: draft-quic-atsss-reqs.mkd [hunk 6]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
 
+<<<<<<< Updated upstream
 Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, round-trip-time estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, round-trip-time, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together. This option is not considered as a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
+=======
+Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, round-trip-time estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, round-trip-time, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, RTT estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, RTT, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together. 
```

File: draft-quic-atsss-reqs.mkd [hunk 7]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
 
 More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead might be considered, if needed. 
+=======
+In 3GPP architectures, a variety of encapsulations are already used to carry user data. The use of QUIC as a tunneling method for ATSSS purposes will add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given that an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. This compares unfavorably to ATSSS Phase 1, which does not add any overhead. 
+
+More investigation is required to assess whether the ATSSSv2 overhead is an issue. Solutions to optimize that overhead could be considered, if needed.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead could be considered, if needed. 
```

**Review by boucadair — CHANGES_REQUESTED** [review]
Thanks for doing this.

**Review by boucadair — APPROVED** [review]

File: draft-quic-atsss-reqs.mkd [hunk 8]
```diff
@@ -428,94 +428,94 @@ Legend:
...
+but splitting cannot be supported. 
+
+Path validation
+induces a delay when switching as packets are buffered. Further, congestion control state is
+reset on a new path and needs to ramp up slowly. When frequent handovers happen, 
 splitting traffic over multiple paths simulaneously can be beneficial for a smooth user experience.
 Further, of course, the ability to use multiple paths simultaneously also increases the 
 maximum capacity available, which can also be benefical in some cases.
 
+<<<<<<< Updated upstream
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 9]
```diff
@@ -428,94 +428,94 @@ Legend:
...
 This option is not a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 10]
```diff
@@ -428,94 +428,94 @@ Legend:
...
+=======
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 11]
```diff
@@ -428,94 +428,94 @@ Legend:
...
+This option is not considered a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
+>>>>>>> Stashed changes
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 12]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
 a split flow. If reordering should be avoided, this would require additional signaling
-between the UE and the ATSSS UPF, e.g., adding sequence numbers, and a reordering buffer and logic.
+between the UE and the ATSSS UPF, e.g., adding sequence numbers, and adding a reordering buffer and logic to both the UE and ATSSS UPF.
 
 Furthermore, since the bandwidth
 available for each QUIC session varies as a function of the congestion experienced
 over each access network, data sent over a congested session could delay the
 delivery of subsequent data over the other session. 
 
+<<<<<<< Updated upstream
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 13]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
 Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, round-trip-time estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, round-trip-time, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together. This option is not considered as a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
+=======
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 14]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
+Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, RTT estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, RTT, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together.
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 15]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
 
+<<<<<<< Updated upstream
 Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, round-trip-time estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, round-trip-time, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together. This option is not considered as a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
+=======
+Experience with MPTCP on smartphones shows that its integrated mechanisms (e.g., congestion control, RTT estimation, packet scheduler) are well suited to support splitting and switching. Providing a similar service over independent QUIC connections would require a complex application that would need to track the congestion window, RTT, and loss characteristics of the underlying QUIC connections as well as some specific application layer signalling to glue the various QUIC connections together. 
+>>>>>>> Stashed changes
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 16]
```diff
@@ -526,22 +526,28 @@ However, this approach for splitting is challenging
...
 
-### MP-QUIC Tunneling
+This option is not considered a valid approach for the ATSSSv2 discussed within the 3GPP.
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
This option is not considered a valid approach for the ATSSS Phase 2 discussed within the 3GPP.
```

File: draft-quic-atsss-reqs.mkd [hunk 17]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
 (2) QUIC Tunnel is a close proposal providing similar
 functionalities to MASQUE based on a binary protocol {{I-D.piraux-quic-tunnel}}{{I-D.piraux-quic-tunnel-tcp}}, and focusing on the ATSSS use case. 
 
-Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-data}}.
+Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-single-tunnel}} and {{sec-quic-multiple-tunnels}}.
 
 ## Encapsulation Overhead {#sec-overhead}
 
+<<<<<<< Updated upstream
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 18]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
 In 3GPP architectures, a variety of encapsulations are already used to convey user data. The use of QUIC as a tunneling method for ATSSS purposes will thus add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given than an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. As a reminder, ATSSS Phase 1 does not add any overhead.
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 19]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 20]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
 
-Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-data}}.
+Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-single-tunnel}} and {{sec-quic-multiple-tunnels}}.
 
 ## Encapsulation Overhead {#sec-overhead}
 
+<<<<<<< Updated upstream
 In 3GPP architectures, a variety of encapsulations are already used to convey user data. The use of QUIC as a tunneling method for ATSSS purposes will thus add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given than an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. As a reminder, ATSSS Phase 1 does not add any overhead. 
 
 More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead might be considered, if needed.
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 21]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
-Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-data}}.
+Both proposals rely upon single QUIC connections and inherit thus the same issues discussed in {{sec-quic-single-tunnel}} and {{sec-quic-multiple-tunnels}}.
 
 ## Encapsulation Overhead {#sec-overhead}
 
+<<<<<<< Updated upstream
 In 3GPP architectures, a variety of encapsulations are already used to convey user data. The use of QUIC as a tunneling method for ATSSS purposes will thus add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given than an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. As a reminder, ATSSS Phase 1 does not add any overhead. 
 
 More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead might be considered, if needed. 
+=======
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

File: draft-quic-atsss-reqs.mkd [hunk 22]
```diff
@@ -613,99 +618,26 @@ Another use case in scope for MASQUE is IP proxying which can be used for tunnel
...
+In 3GPP architectures, a variety of encapsulations are already used to carry user data. The use of QUIC as a tunneling method for ATSSS purposes will add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given that an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. This compares unfavorably to ATSSS Phase 1, which does not add any overhead. 
+
+More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead could be considered, if needed. 
+>>>>>>> Stashed changes
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
```

**Review by mpiraux — COMMENTED** [review]
Thank you for doing this. I just fixed some stashed changes that were not merged yet.

**Pull request merged**